lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2768,6 +2770,34 @@ public function createEntity($className, array $data, &$hints = array())
     }
 
     /**
+     * @param array $data
+     * @param ClassMetadata $class
+     */
+    private function removeNullableEmbeddedReferences(array &$data, ClassMetadata $class)
```

Comment:
@Ocramius, @1ed and @beberlei what do you think to extract that logic from ```UnitOfWork``` into ```ClassMetadata```? It seems to be proper location, considering that ```UnitOfWork``` is already huge.

I thought on something like ```ClassMetadata::updateEntity($entity, array $data)``` or  ```ClassMetadata::applyData($entity, array $data)```.